packages/protocol/src/vaults/borrowing/BorrowingVault.sol

```diff
@@ -613,7 +613,7 @@ contract BorrowingVault is BaseVault {
     _payback(caller, owner, debtToCover, debtSharesToCover);
 
     // Ensure liquidator receives no more shares than 'owner' owns.
-    uint256 existingShares = balanceOf(owner);
+    uint256 existingShares = maxRedeem(owner);
```

Comment:
I would say the maxRedeem makes more sense than the balanceOf. This line is executed after the payback's been done. So, the paid debt will release a certain amount of collateral that may not always be all the user's balance (sometimes it is less).
The check with balanceOf was not passing the tests because of the require in the image. When the amount of collateral released by the paid debt was less then the balance of the user, the require in the image triggered.
<img width="640" alt="Screenshot 2023-04-04 at 10 52 30" src="https://user-images.githubusercontent.com/80268365/229674940-9695e2c0-bfd4-4235-9488-32362ebcb858.png">
I figured it would be coherent to have the same check done in the liquidate function because it is always going to verify against the maxRedeem and not the balance.

Reply:
I resolved the issue in _beforeTokenTransfer in pull request #402 which I just added you as a checker now that you are in the subject 😄